source/scpd/quorum/QuorumIntersectionChecker.d

```diff
@@ -0,0 +1,42 @@
+/*******************************************************************************
+
+    Bindings for quorum/QuorumIntersectionChecker.h
+
+    Copyright:
+        Copyright (c) 2020 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module scpd.quorum.QuorumIntersectionChecker;
+
+nothrow @trusted @nogc:
+
+import scpd.Cpp;
+import scpd.types.Stellar_types;
+import scpd.quorum.QuorumTracker;
+
+extern (C++, `stellar`):
+
+/// Ditto
+extern(C++, class) public abstract class QuorumIntersectionChecker
+{
+  public:
+    /// Create & initialize a QuorumIntersectionChecker with the given map
+    static shared_ptr!QuorumIntersectionChecker create (
+        ref const(QuorumTracker.QuorumMap) map);
+
+    ~this ();
+
+    /// Returns: true if the network enjoys quorum intersection
+    abstract bool networkEnjoysQuorumIntersection ();
+
+    /// Returns: the total number of quorums found
+    abstract size_t getMaxQuorumsFound ();
+
+    /// Returns: A pair of possible quorum splits found, or empty pair if none
+    abstract pair!(vector!NodeID, vector!NodeID) getPotentialSplit ();
+}
```

Comment:
Sure, but why?

Reply:
Why it is `8` or why we should have this check ?

Reply:
@AndrejMitrovic answered directly: the later.

Because it's a rather simple check that can save you a lot of headache. For example, out of the two types introduced in this PR, one had the wrong size (`QuorumTracker`) and adding the check uncovered that and #753 . Also makes sure that in the future, we don't break this by accident (e.g. change in alignment, or adding a field in a nested type).

In my experience, those little checks have an immediate benefit.